test/cartesian/CartesianGrid.spec.tsx

```diff
@@ -310,28 +483,26 @@ describe('<CartesianGrid />', () => {
           ticks: ['x', 'y', 'x'],
         };
         render(
-          <Surface width={500} height={500}>
+          <AreaChart width={300} height={200} margin={chartMargin}>
```

Comment:
So, is this a breaking change? This case no longer works with the Surface?

Reply:
The CategoricalChart will no longer render on its own, it now requires a context coming from one of the generateCategoricalChart.

Do you think it's a breaking change? It should be fairly easy to fix (few null checks will do the trick), I just never expected that to be a use-case.

Reply:
Well actually it still does render independently (or with Surface, to get the svg tag parent) in this pull request. But it will break in the next pull request.

Reply:
I agree that this use case is not intended, but I want to raise awareness to breaking changes. If this tests fails with the surface, it strictly speaking is a breaking change - that being said: One that I am willing to accept for moving forward with improved state management. The benefits do outweigh the cost imho. 

But others might disagree. @ckifer

Reply:
Hmm, not sure how I feel. The components that don't need context should probably render without it if all the right data is supplied. I could see people rendering things separately or doing some sort of really custom thing with the recharts building block components. 

But I agree in principle, we should move forward else we'll just keep existing where we are

Reply:
Does our website or examples show or promote usage of individual components? Most of them won't work alone anyway because of their dependency on generateCategoricalChart state. So I assumed that's a design decision.

If we do want to support individual rendering: we will document and export the ChartLayoutContext. Then we get maximum flexibility and anyone can rendering everything without gCC.

Perhaps we should start another discussion and see if that's valuable for our users?

Reply:
So I know that people have used shapes on their own (Rectangle, Curve, etc.). `Text` probably too.

Others that I'm not sure about but may be useful to render without a greater context:

* Legend
* Label (?) 

Everything else is probably ok as long as they do render _individually_ within a chart component.

ex: I want this to render an XAxis without a chart

```
<ComposedChart height={200} width={200}>
  <XAxis {...props I need for axis to render} >
</ComposedChart
```